docs/metrics.md

```diff
@@ -0,0 +1,97 @@
+scp.pending.discarded             | counter   | number of discarded envelopes
+scp.pending.fetching              | counter   | number of incomplete envelopes
+scp.pending.ready                 | counter   | number of envelopes ready to process
+history.publish.success           | meter     | published completed successfuly
+history.publish.failure           | meter     | published failed
+history.download-<X>.success      | meter     | download of <X> completed successfuly
+history.download-<X>.failure      | meter     | download of <X> failed
+history.verify-<X>.success        | meter     | verification of <X> succeeded
+history.verify-<X>.failure        | meter     | verification of <X> failed
+invariant.does-not-hold.count.<X> | counter   | number of times invariant <X> failed
+ledger.transaction.apply          | timer     | time to apply one transaction
+ledger.transaction.count          | histogram | number of transactions per ledger
+ledger.ledger.close               | timer     | time to close a ledger (excluding consensus)
+ledger.age.closed                 | timer     | time between ledgers
+ledger.age.current-seconds        | counter   | gap between last close ledger time and current time
+ledger.memory.queued-ledgers      | counter   | number of ledgers queued in memory for replay
+app.state.current                 | counter   | state (BOOTING=0, JOIN_SCP=1, LEDGER_SYNC=2, CATCHING_UP=3, SYNCED=4, STOPPING=5)
+overlay.memory.flood-known        | counter   | number of known flooded entries
+overlay.flood.broadcast           | meter     | message sent as broadcast per peer
+overlay.message.broadcast         | meter     | message broadcasted
+overlay.connection.outbound-start | meter     | outbound connection initiated
+overlay.connection.establish      | meter     | connection established (pending inbound/outbound)
+overlay.connection.drop           | meter     | connection dropped
+overlay.connection.reject         | meter     | connection rejected
+overlay.connection.pending        | counter   | number of pending connections
+overlay.connection.authenticated  | counter   | number of authenticated peers
+overlay.byte.read                 | meter     | number of bytes received
+overlay.byte.write                | meter     | number of bytes sent
+overlay.message.read              | meter     | message received
+overlay.message.write             | meter     | message sent
+overlay.error.read                | meter     | error while receiving a message
+overlay.error.write               | meter     | error while sending a message
+overlay.timeout.idle              | meter     | idle peer timeout
+overlay.recv.<X>                  | timer     | received message <X>
+overlay.send.<X>                  | meter     | sent message <X>
+overlay.item-fetcher.next-peer    | meter     | ask for item past the first one
+loadgen.step.count                | meter     | loadgenerator step
```

Comment:
The description sounds a bit vague for anyone without the context of load generator. Can we say something like "number of steps to generate requested load"?